UICatalog/Scenarios/ColorPicker.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Terminal.Gui;
+using System;
+
+namespace UICatalog.Scenarios {
+	[ScenarioMetadata (Name: "Color Picker Demo", Description: "Color Picker Demo.")]
+	[ScenarioCategory ("Colors")]
+	public class ColorPickers : Scenario {
+		/// <summary>
+		/// Foreground ColorPicker.
+		/// </summary>
+		private ColorPicker foregroundColorPicker;
+
+		/// <summary>
+		/// Background ColorPicker.
+		/// </summary>
+		private ColorPicker backgroundColorPicker;
+
+		/// <summary>
+		/// Foreground color label.
+		/// </summary>
+		private Label foregroundColorLabel;
+
+		/// <summary>
+		/// Background color Label.
+		/// </summary>
+		private Label backgroundColorLabel;
+
+		/// <summary>
+		/// Demo label.
+		/// </summary>
+		private Label demoLabel;
+
+		/// <summary>
+		/// Setup the scenario.
+		/// </summary>
+		public override void Setup ()
+		{
+			// Scenario Window's.
+			Win.Title = this.GetName ();
+
+			// Forground ColorPicker.
+			foregroundColorPicker = new ColorPicker (1, 1, "Foreground color");
+			foregroundColorPicker.ColorChanged += ForegroundColor_ColorChanged;
+			Win.Add (foregroundColorPicker);
+
+			foregroundColorLabel = new Label (1, 10, string.Empty);
+			Win.Add (foregroundColorLabel);
+
+			// Background ColorPicker.
+			backgroundColorPicker = new ColorPicker (85, 1, "Background color");
```

Comment:
I noticed that if my terminal is not default width, the controls in the Scenario get clipped etc...

I guessed you had not tested using Relative Layout.

I was right. The control is completely broken when using Relative Layout.

Please:

1) Change the Scenario to demonstrate that both Absolute and Relative Layout work.
2) Fix whatever is wrong that prevents #1 from currently working ;-).

Here's code that SHOULD align the color picker to the right of the window:

```csharp
// Background ColorPicker.
backgroundColorPicker = new ColorPicker ();
backgroundColorPicker.Text = "Background Color Pickeroo";
backgroundColorPicker.X = Pos.AnchorEnd () - (Pos.Right (backgroundColorPicker) - Pos.Left (backgroundColorPicker));
backgroundColorPicker.Y = 1;
backgroundColorPicker.ColorChanged += BackgroundColor_ColorChanged;
Win.Add (backgroundColorPicker);
```

Reply:
The component manager now the relative layout, but I have a problem this borders. If I set a border, the size of the component doesn't count the border. Is it normal ? So the relative layout not working well.
Can you explain me, please what I have to do with high-intensity colors ?

Reply:
Yes it's normal.. If you want to use borders on your view then you must have two options. With the `PanelView` you have the possibility of having your view growing with the borders or maintaining always the same size by shrinking inside, dependent on the `UsePanelFrame` property. Without the `PanelView` your view can only be growing within the borders, unless it's implemented as `Window` or `FrameView`. If you want to measure the total borders size you can use the `GetSumThickness` method of the `Border` class. Please see the `Borders` scenario to learn about the both implementations and differences.
With the high-intensity colors I have no idea yet.

Reply:
> Can you explain me, please what I have to do with high-intensity colors ?

I had typed `#1` and github expanded that to Issue # 1, which is not what I meant.  I was referring to my 1st point which was `Change the Scenario to demonstrate that both Absolute and Relative Layout work.`.  My apologies for not noticing github had expanded `#1` like this and confusing you!

I see you've already updated the scenario to use relative positioning, so that part is good!

Reply:
> Yes it's normal.. If you want to use borders on your view then you must have two options. With the `PanelView` you have the possibility of having your view growing with the borders or maintaining always the same size by shrinking inside, dependent on the `UsePanelFrame` property. Without the `PanelView` your view can only be growing within the borders, unless it's implemented as `Window` or `FrameView`. If you want to measure the total borders size you can use the `GetSumThickness` method of the `Border` class. Please see the `Borders` scenario to learn about the both implementations and differences. With the high-intensity colors I have no idea yet.

Thank you for explaining. At the end I think it's better without borders by default and the user can choose to add or not depending if he wants use the control in a (Dialog, PanelView) or not. In fact, the relative layout works well without borders. With border maybe it's normal to taking the borders width in account when positioning the control in the page. Is it right ? I saw the same behaviour with other controls.

Reply:
> > Can you explain me, please what I have to do with high-intensity colors ?
> 
> I had typed `#1` and github expanded that to Issue # 1, which is not what I meant. I was referring to my 1st point which was `Change the Scenario to demonstrate that both Absolute and Relative Layout work.`. My apologies for not noticing github had expanded `#1` like this and confusing you!
> 
> I see you've already updated the scenario to use relative positioning, so that part is good!

Ahhh ok. No problem :-)